pkg/distributor/distributor.go

```diff
@@ -864,27 +907,10 @@ func (d *Distributor) PushWithCleanup(ctx context.Context, req *mimirpb.WriteReq
 	// For each timeseries, compute a hash to distribute across ingesters;
 	// check each sample and discard if outside limits.
 	for _, ts := range req.Timeseries {
-		if mrc := d.limits.MetricRelabelConfigs(userID); len(mrc) > 0 {
-			l := relabel.Process(mimirpb.FromLabelAdaptersToLabels(ts.Labels), mrc...)
-			ts.Labels = mimirpb.FromLabelsToLabelAdapters(l)
-		}
-
-		for _, labelName := range d.limits.DropLabels(userID) {
-			removeLabel(labelName, &ts.Labels)
-		}
-
 		if len(ts.Labels) == 0 {
 			continue
 		}
 
-		// We rely on sorted labels in different places:
```

Comment:
Why is sorting of labels moved to `prePushRelabelMiddleware`? Is it to make sure we forward correct (sorted) labels?

Reply:
Yes, because other middlewares after this one or the final push method might rely on the labels being sorted. 
if a relabel rule gets applied on a labelset then the result should always be sorted, but there is no guarantee that a relabel rule gets applied on every series, so I figured it would make sense to sort here because that way we can be sure that after the relabel middleware all labelsets are guaranteed to be sorted.
Do you think we should move it?

Reply:
I think we should move it to "validation" middleware at some point, but let's keep it here for now.

Reply:
I was mostly curious if I understand the reasoning.

Reply:
> I think we should move it to "validation" middleware at some point, but let's keep it here for now.

that makes sense to me once we have the validation middleware